### PR TITLE
Docker: Add docker container under subdir with required packages

### PR DESCRIPTION
**Pull request by StaticRocket**
Use my existing container for this since it handles ID remapping automatically and has the required tools to run all the linting scripts under `bin/`.

Adds a workflow with max caching enabled to save some time.

https://github.com/StaticRocket/psdk-doc-docker

closes: #5

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# SPDX-License-Identifier: GPL-2.0
+
```
[the remark this answers is not in the document]
Comment (nmenon) [reply docker/Dockerfile:2]: Use SPDX instead of full copy paste

File: docker/Makefile [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# SPDX-License-Identifier: GPL-2.0
+
+# Copyright (C) 2024 Randolph Sapp
+#
+# This program is free software; you can redistribute it and/or modify it under
```
Comment (nmenon) [inline docker/Makefile:5]: Same problem here as well

File: docker/root/init [hunk 3]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: GPL-2.0
```
Comment (nmenon) [inline docker/root/init:3]: same issue

File: README.md [hunk 4]
```diff
@@ -5,15 +5,13 @@ Processor SDK Documentation in Sphinx
...
-OR use the following command in a python virtual environment for a known working
-config as of 2023/09/05:
+Use the following command in a python virtual environment for a known working
+config:
 
     # python3 -m pip install -r requirements.txt
 
 OR you can use a docker container like the following:
-    - [psdk-doc-docker](https://github.com/StaticRocket/psdk-doc-docker/pkgs/container/psdk-doc-docker)
+    - [psdk-doc-docker](https://github.com/TexasInstruments/processor-sdk-doc/pkgs/container/processor-sdk-doc)
```
Comment (nmenon) [inline README.md:14]: why not link this to the readme.md file we are just adding?

File: docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+		--no-install-recommends \
+		dumb-init \
+		git \
+		gosu \
+		make \
+		python3-pip \
+		ripgrep \
+		zip \
+	&& echo "**** create abc user and make our folders ****" \
+	&& useradd -u 1000 -U -d /config -s /bin/false abc \
```
Comment (nmenon) [inline docker/Dockerfile:31]: make the userID an arg - there are users with kerberos whose user ID != 1000 - they should at least be able to build OR you would need to handle that as part of a wrapper script

File: .github/workflows/docker.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
+---
+name: Docker Workflow
+on:
+  push:
+    branches: [master]
```
Comment (nmenon) [inline .github/workflows/docker.yml:5]: do we want to rebuild the container for every document push to master?

File: docker/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, see <https://www.gnu.org/licenses/>.
+
+FROM debian:stable-slim
```
Comment (nmenon) [inline docker/Dockerfile:18]: Also add:
LABEL org.opencontainers.image.source https://github.com/TexasInstruments/processor-sdk-doc

File: requirements.txt [hunk 8]
```diff
@@ -21,3 +21,4 @@ sphinxcontrib-jsmath==1.0.1
 sphinxcontrib-qthelp==1.0.3
 sphinxcontrib-serializinghtml==1.1.5
 urllib3==2.2.1
+rstcheck==3.3.1
```
Comment (nmenon) [inline requirements.txt:24]: should have been a different commit

File: docker/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+RUN apt-get update \
+	&& DEBIAN_FRONTEND=noninteractive apt-get install -y \
+		--no-install-recommends \
+		dumb-init \
+		git \
+		gosu \
```
Comment (nmenon) [inline docker/Dockerfile:25]: why gosu ?

File: docker/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+		make \
+		python3-pip \
+		ripgrep \
+		zip \
```
Comment (nmenon) [inline docker/Dockerfile:29]: these packages must be listed in README.md as well.

File: README.md (hunk 4 above)
Comment (StaticRocket) [reply README.md:14]: This links to the page that will be created on first run that shows all tags and the README in one place.

File: docker/Dockerfile (hunk 5 above)
Comment (StaticRocket) [reply docker/Dockerfile:31]: You didn't look at the ID remap script did you...

File: .github/workflows/docker.yml (hunk 6 above)
Comment (StaticRocket) [reply .github/workflows/docker.yml:5]: That's precisely why the results are cached in the workflow and why I called it out explicitly in the PR and commit.

File: docker/Dockerfile (hunk 7 above)
Comment (StaticRocket) [reply docker/Dockerfile:18]: This label is added automatically by the workflow metadata script. Along with a few other useful labels.

File: docker/Dockerfile (hunk 9 above)
Comment (StaticRocket) [reply docker/Dockerfile:25]: Required for ID remapping as su / sudo regularly eat extra arguments and misbehave under docker.

File: docker/Dockerfile (hunk 10 above)
Comment (StaticRocket) [reply docker/Dockerfile:29]: These packages are check dependencies required for our scripts. They are not build requirements.

File: README.md (hunk 4 above)
Comment (nmenon) [reply README.md:14]: I meant put that information in the docker/readme.md and point using :ref: to that doc

File: docker/Dockerfile (hunk 5 above)
Comment (nmenon) [reply docker/Dockerfile:31]: where is the script?

File: docker/root/init [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, see <https://www.gnu.org/licenses/>.
+
+get_attribs() {
+	local file_stats file_to_test useful_attribs
+	if file_to_test=$(realpath "$1") && [[ $2 =~ ^[0-9]+$ ]] ; then
+		useful_attribs=$(stat "$file_to_test" -t)
+		read -r -a file_stats <<< "${useful_attribs#"$file_to_test"}"
+		echo "${file_stats["$2"]}"
```
Comment (nmenon) [inline docker/root/init:25]: you are assuming file uid is equal to user's id?

**Review by nmenon — CHANGES_REQUESTED** [review]

File: docker/root/init (hunk 11 above)
Comment (StaticRocket) [reply docker/root/init:10]: Yeah. It's what things like the Yocto build containers use to make sure permissions between the host and container are in sync.

In the case where a user needs to override this behavior, they can simply set the entrypoint to bash and specify uid:gid as normal.

File: docker/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: MIT
+
```
Comment (nmenon) [inline docker/Dockerfile:2]: Missing:
Copyright (C) 2024 Texas Instruments Incorporated - https://www.ti.com

File: docker/Makefile [hunk 13]
```diff
@@ -0,0 +1,22 @@
+# SPDX-License-Identifier: MIT
+
```
Comment (nmenon) [inline docker/Makefile:2]: Missing:
Copyright (C) 2024 Texas Instruments Incorporated - https://www.ti.com

File: docker/root/init [hunk 14]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: MIT
+
```
Comment (nmenon) [inline docker/root/init:4]: Missing:
Copyright (C) 2024 Texas Instruments Incorporated - https://www.ti.com

**Review by nmenon — CHANGES_REQUESTED** [review]
Minor updates pending..

File: docker/root/init (hunk 11 above)
Comment (nmenon) [reply docker/root/init:10]: OK - Thanks.

**Comment by nmenon** [comment]
Other than copyright, things look fine and good to go from my side.

**Review by nmenon — APPROVED** [review]
LGTM. Thanks

**Review by cshilwant — APPROVED** [review]
LGTM!

**Review by praneethbajjuri — APPROVED** [review]

**Pull request merged**